packages/react-instantsearch-core/src/core/createConnector.tsx

```diff
@@ -0,0 +1,385 @@
+          'We are now provided a dedicated API' +
+          ' the `connectStateResults` connector that you should use instead. The `createConnector` API will be ' +
+          'soon deprecated and will break in future next major versions.' +
+          '\n\n' +
+          'See more at https://www.algolia.com/doc/api-reference/widgets/state-results/react/' +
+          '\n' +
+          'and https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-display/react/'
+      );
+    }
+  }
+}
+
+export type ConnectorDescription = {
+  displayName: string;
+  /**
+   * a function to filter the local state
+   */
+  refine?: (...args: any[]) => any;
+  /**
+   * function transforming the local state to a SearchParameters
+   */
+  getSearchParameters?: (...args: any[]) => any;
+  /**
+   * metadata of the widget (for current refinements)
+   */
+  getMetadata?: (...args: any[]) => any;
+  /**
+   * hook after the state has changed
+   */
+  transitionState?: (...args: any[]) => any;
+  /**
+   * transform the state into props passed to the wrapped component.
+   * Receives (props, widgetStates, searchState, metadata) and returns the local state.
+   */
+  getProvidedProps: (...args: any[]) => any;
+  /**
+   * Receives props and return the id that will be used to identify the widget
+   */
+  getId?: (...args: any[]) => string;
+  /**
+   * hook when the widget will unmount. Receives (props, searchState) and return a cleaned state.
+   */
+  cleanUp?: (...args: any[]) => any;
+  searchForFacetValues?: (...args: any[]) => any;
+  shouldComponentUpdate?: (...args: any[]) => boolean;
+  /**
+   * PropTypes forwarded to the wrapped component.
+   */
+  propTypes?: {}; // I can't find a definition for a propTypes object
+  defaultProps?: {};
+};
+
+/**
+ * Connectors are the HOC used to transform React components
+ * into InstantSearch widgets.
+ * In order to simplify the construction of such connectors
+ * `createConnector` takes a description and transform it into
+ * a connector.
+ * @param {ConnectorDescription} connectorDesc the description of the connector
+ * @return {Connector} a function that wraps a component into
+ * an instantsearch connected one.
+ */
+export function createConnectorWithoutContext(
+  connectorDesc: ConnectorDescription
+) {
+  if (!connectorDesc.displayName) {
+    throw new Error(
+      '`createConnector` requires you to provide a `displayName` property.'
+    );
+  }
+
+  needlessUsageWarning(connectorDesc);
+
+  const isWidget =
+    typeof connectorDesc.getSearchParameters === 'function' ||
+    typeof connectorDesc.getMetadata === 'function' ||
+    typeof connectorDesc.transitionState === 'function';
+
+  type ConnectorProps = {
+    contextValue: InstantSearchContext;
+  };
+
+  type ConnectorState = {
+    providedProps: {};
+  };
+
+  return (Composed: ReactType) => {
+    class Connector extends Component<ConnectorProps, ConnectorState> {
```

Comment:
At the moment it's fine but the types are not correct, it should be something like this:

https://github.com/algolia/react-instantsearch/blob/fd76ff767934c71e9f8560ca01d3795e4946b0a9/packages/react-instantsearch-dom-maps/src/withGoogleMaps.tsx#L15-L23

Reply:
I have something more like that in step 7